src/Rule/Url.php

```diff
@@ -63,9 +63,11 @@ protected function validateValue($value, ValidationContext $context = null): Res
         return $result;
     }
 
-    private function idnToAscii($idn)
+    private function idnToAscii($idn): string
```

Comment:
```suggestion
    private function idnToAscii(string $idn): string
```
Should $id be string?

Reply:
Also add type to parameter of method `convertIdn()`.